Microsoft.Azure.Cosmos/src/Query/Core/CoordinatorDistributionPlan/ClientQLDeserializer.cs

```
@@ -0,0 +1,565 @@
+        {
+            ClientQLUserDefinedFunctionCallScalarExpression userDefinedFunctionCallExpression = new ClientQLUserDefinedFunctionCallScalarExpression
+            {
+                Kind = ClientQLScalarExpressionKind.UserDefinedFunctionCall,
+                Identifier = token["Identifier"].ToObject<ClientQLFunctionIdentifier>(serializer),
+                VecArguments = token["VecArguments"].ToObject<List<ClientQLScalarExpression>>(serializer),
+                Builtin = token.Value<bool>("Builtin")
+            };
+
+            return userDefinedFunctionCallExpression;
+        }
+
+        public ClientQLVariableRefScalarExpression DeserializeVariableRefScalarExpression(JToken token, JsonSerializer serializer)
+        {
+            ClientQLVariableRefScalarExpression variableRefExpression = new ClientQLVariableRefScalarExpression
+            {
+                Kind = ClientQLScalarExpressionKind.VariableRef,
+                Variable = this.DeserializeClientQLVariable(token["Variable"])
+            };
+
+            return variableRefExpression;
+        }
+
+        private List<ClientQLOrderByItem> DeserializeOrderByItems(JToken token, JsonSerializer serializer)
+        {
+            List<ClientQLOrderByItem> orderByItems = new List<ClientQLOrderByItem>();
+
+            if (token != null && token.Type == JTokenType.Array)
+            {
+                foreach (JToken itemToken in token)
+                {
+                    ClientQLOrderByItem orderByItem = new ClientQLOrderByItem
+                    {
+                        Expression = this.DeserializeScalarExpression(itemToken["Expression"], serializer),
+                        SortOrder = (ClientQLSortOrder)Enum.Parse(typeof(ClientQLSortOrder), itemToken.Value<string>("SortOrder"))
+                    };
+
+                    orderByItems.Add(orderByItem);
+                }
+            }
```

>} [](http://example.com/codeflow?start=12&length=1)

Please fail fast and early in the else case. Here and throughout the deserializer execution.